go-controller/pkg/util/multi_network.go

```diff
@@ -682,17 +682,34 @@ func NewNetInfo(netconf *ovncnitypes.NetConf) (NetInfo, error) {
 	if netconf.Name == types.DefaultNetworkName {
 		return &DefaultNetInfo{}, nil
 	}
+	var ni NetInfo
+	var err error
 	switch netconf.Topology {
 	case types.Layer3Topology:
-		return newLayer3NetConfInfo(netconf)
+		ni, err = newLayer3NetConfInfo(netconf)
 	case types.Layer2Topology:
-		return newLayer2NetConfInfo(netconf)
+		ni, err = newLayer2NetConfInfo(netconf)
 	case types.LocalnetTopology:
-		return newLocalnetNetConfInfo(netconf)
+		ni, err = newLocalnetNetConfInfo(netconf)
 	default:
 		// other topology NAD can be supported later
 		return nil, fmt.Errorf("topology %s not supported", netconf.Topology)
 	}
+	if err != nil {
+		return nil, err
+	}
+
+	// secondary networks can support ipv4 and ipv6 regardless of the underlying cluster
+	if !ni.IsSecondary() {
```

Comment:
@JacobTanenbaum I think a primary network also returns true here ...

IIUC, we would want to return true **only** for secondary nets - i.e. `isSecondary && role == primary`. 

... I know, the name for secondary is **very** misleading. I think @trozet meant to push something to address this ?..

Reply:
added the check for `IsPrimary()` would that work?